packages/app/src/providers/utils.ts

```diff
@@ -94,3 +76,28 @@ export async function getValueOrError(
     return error;
   }
 }
+
+export function processAxiosError(
+  error: unknown,
+  getMessageToThrow: (error: AxiosError) => string | false | undefined,
```

Comment:
It feels weird to have to pass a callback when you already have access to the argument `error` in the caller. It makes the calling code way less readable IMO.

Perhaps pass `messageToThrow` directly and have it `undefined` if `error` is not an `AxiosError` ?

Reply:
The goal was to abstract away the condition `error instanceof AxiosError`, but I agree that the callback is not super readable. In this case, I prefer to remove the abstraction altogether and inline the error handling in the provider APIs.

This is where wretch's [chainable API](https://github.com/elbywan/wretch?tab=readme-ov-file#catchers-) would really shine. Perhaps I will migrate to it at some point.

Reply:
I've pushed a separate commit: https://github.com/silx-kit/h5web/pull/1694/commits/47a4e9cc159726f42d66985af123e29a76e84adc

Reply:
Excellent, that's way clearer